src/ChunkMap.cpp

```diff
@@ -44,99 +44,35 @@ cChunkMap::cChunkMap(cWorld * a_World) :
 		)
 	)
 {
-
-}
-
-
-
-
-
-cChunkMap::~cChunkMap()
-{
-	cCSLock Lock(m_CSLayers);
-	while (!m_Layers.empty())
-	{
-		delete m_Layers.back();
-		m_Layers.pop_back();  // Must pop, because further chunk deletions query the chunkmap for entities and that would touch deleted data
```

Comment:
Its complicated. The default destructor won't take the lock whilst deleting the map, but I think its UB if anything else is touching the world whilst its being deleted, so that shouldn't matter.